d.ts/flv.d.ts

```diff
@@ -141,39 +244,89 @@ declare namespace FlvJs {
     }
 
     interface Player {
-        constructor: PlayerConstructor;
         destroy(): void;
-        on(event: string, listener: () => void): void;
-        off(event: string, listener: () => void): void;
+        on(event: string, listener: (...args: any[]) => void): void;
+        off(event: string, listener: (...args: any[]) => void): void;
         attachMediaElement(mediaElement: HTMLMediaElement): void;
         detachMediaElement(): void;
         load(): void;
         unload(): void;
-        play(): Promise<void>;
+        play(): Promise<void> | void;
         pause(): void;
         type: string;
         buffered: TimeRanges;
         duration: number;
         volume: number;
         muted: boolean;
         currentTime: number;
-        mediaInfo: AnyObject;
-        statisticsInfo: AnyObject;
+        /**
+         * @deprecated FlvPlayer/NativePlayer have its own `mediaInfo` field.
+         * @desc Keep it for backwards compatibility
+         * @since 1.4
+         */
+        mediaInfo: NativePlayerMediaInfo | FlvPlayerMediaInfo;
+        /**
+         * @deprecated FlvPlayer/NativePlayer have its own `statisticsInfo` field.
+         * @desc Keep it for backwards compatibility
+         * @since 1.4
+         */
+        statisticsInfo: NativePlayerStatisticsInfo | FlvPlayerStatisticsInfo;
     }
 
-    /**
-     * @deprecated Use FlvJs.Player instead.
-     */
-    // tslint:disable-next-line
-    interface FlvPlayer extends Player {}
+    interface NativePlayerStatisticsInfo {
+        playerType: 'NativePlayer';
+        url: string;
+        decodedFrames?: number;
+        droppedFrames?: number;
+    }
 
-    /**
-     * @deprecated Use FlvJs.Player instead.
-     */
-    // tslint:disable-next-line
-    interface NativePlayer extends Player {}
+    interface FlvPlayerReportStatisticsInfo {
+        url: string;
+        hasRedirect: boolean;
+        redirectedURL?: string;
```

Comment:
`redirectedURL` is an optional field which exists only if `hasRedirect === true`.
`redirectedURL?: string | undefined` may be better?

Reply:
I think `redirectedURL?: string`  equal to `redirectedURL: string | undefined`.  Optional field indicate that `redirectedURL` maybe `undefined`.

Reply:
Oh I didn't notice the `?`